### PR TITLE
(tree): remove internal tree types from public export

### PR DESCRIPTION
**Pull request by zhenmichael**
Reference for list of API Issues:
https://github.com/microsoft/FluidFramework/pull/19724/files

Some of the tree types in the current public export involve type manipulation which shouldn't be exposed. These types are mainly used by other exported APIs, so they are now exported from an internal types module instead.

### CODE REVIEW

**Review by ChumpChief — APPROVED** [review]
I think this looks reasonable - @CraigMacomber can you please also sign off since this is based on your feedback?

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -31595,6 +31595,7 @@ packages:
   /clone@1.0.4:
     resolution: {integrity: sha512-JQHZ2QMW6l3aH/j6xCqQThY/9OH4D/9ls34cgkUBiEeocRTU04tHfKPBsUK1PqZCUQM7GiA0IIXJSuXHI64Kbg==}
     engines: {node: '>=0.8'}
+    requiresBuild: true
```
Comment (CraigMacomber) [inline pnpm-lock.yaml:31598]: the changes to the pnpm-lock file seem unrelated to this PR, and should probably be removed from it.

**Review by CraigMacomber — APPROVED** [review]
We are about to have even more types which we will want to hide away in these namespaces. This is going to really help keep our API clean going forward!
